### PR TITLE
Migrate GoogleCloudPlatform to Config Merger

### PR DESCRIPTION
**Pull request by chases2**
Tracking Issue: https://github.com/GoogleCloudPlatform/oss-test-infra/issues/832

### CODE REVIEW

**Comment by chases2** [comment]
Once this merges, close https://github.com/kubernetes/test-infra/pull/22129

**Review by chaodaiG — APPROVED** [review]
/hold
so it doesn't merge too late in the day

**Pull request merged**
